app/main/posts/detail/post-detail-data.controller.js

```diff
@@ -0,0 +1,222 @@
+module.exports = [
```

Comment:
this is literally grabbing the old directive controller and placing it here instead, with some minor modifications to get state params and such